src/api/schema/gen.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs;
+use vector::api::schema::build_schema;
+
+static INTROSPECTION_QUERY: &str = r#"
+query IntrospectionQuery {
+  __schema {
+    queryType {
+      name
+    }
+    mutationType {
+      name
+    }
+    subscriptionType {
+      name
+    }
+    types {
+      ...FullType
+    }
+    directives {
+      name
+      description
+      locations
+      args {
+        ...InputValue
+      }
+    }
+  }
+}
+fragment FullType on __Type {
+  kind
+  name
+  description
+  fields(includeDeprecated: true) {
+    name
+    description
+    args {
+      ...InputValue
+    }
+    type {
+      ...TypeRef
+    }
+    isDeprecated
+    deprecationReason
+  }
+  inputFields {
+    ...InputValue
+  }
+  interfaces {
+    ...TypeRef
+  }
+  enumValues(includeDeprecated: true) {
+    name
+    description
+    isDeprecated
+    deprecationReason
+  }
+  possibleTypes {
+    ...TypeRef
+  }
+}
+fragment InputValue on __InputValue {
+  name
+  description
+  type {
+    ...TypeRef
+  }
+  defaultValue
+}
+fragment TypeRef on __Type {
+  kind
+  name
+  ofType {
+    kind
+    name
+    ofType {
+      kind
+      name
+      ofType {
+        kind
+        name
+        ofType {
+          kind
+          name
+          ofType {
+            kind
+            name
+            ofType {
+              kind
+              name
+              ofType {
+                kind
+                name
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+"#;
+
+#[tokio::main]
+async fn main() {
+    let schema = build_schema().finish();
+    let res = schema.execute(INTROSPECTION_QUERY).await;
+
+    let json = serde_json::to_string_pretty(&async_graphql::http::GQLResponse(res)).unwrap();
+
+    fs::write("graphql/schema.json", json).expect("Couldn't save schema file");
```

Comment:
Stores the schema at `graphql/schema.json`, based on the cwd. I'm expecting this would typically be invoked via `cargo run` and therefore relative to Cargo.toml. I'm also assuming that the `graphql` dir actually exists, which it should by default (since this file is checked in.)

If we ever want to deploy this separately (say, in a Docker image) or handle cases where the dir structure might not match, then a `mkdir -p` approach + CLI opts would be an obvious upgrade path. Kept this simple for now.